lib/vmpooler/pool_manager.rb

```diff
@@ -761,12 +799,7 @@ def execute!(maxloop = 0, loop_delay = 1)
         end
 
         $config[:pools].each do |pool|
-          if ! $threads[pool['name']]
-            check_pool(pool)
-          elsif ! $threads[pool['name']].alive?
-            $logger.log('d', "[!] [#{pool['name']}] worker thread died, restarting")
-            check_pool(pool)
-          end
+          evaluate_pool(pool, task_limit, maxloop, loop_delay)
```

Comment:
Yes. My perception was that maxloop was there only for testing purposes, and blocking further pool evaluations until a slot is available seems desirable. The current pool manager is obscenely resource intensive on the host system.